src/libmongoc/tests/test-mongoc-structured-log.c

```diff
@@ -699,6 +738,74 @@ test_structured_log_component_names (void)
    ASSERT_CMPSTR (mongoc_structured_log_get_component_name (component), "connection");
 }
 
+void
+test_structured_log_max_document_length (void)
+{
+   mongoc_structured_log_opts_t *opts = mongoc_structured_log_opts_new ();
+
+   ASSERT_CMPINT (MONGOC_STRUCTURED_LOG_DEFAULT_MAX_DOCUMENT_LENGTH, ==, 1000);
+
+   ASSERT (mongoc_structured_log_opts_set_max_document_length (opts, 0));
+   ASSERT (!mongoc_structured_log_opts_set_max_document_length (opts, INT_MAX));
+   ASSERT (mongoc_structured_log_opts_set_max_document_length (opts, INT_MAX / 2));
+   ASSERT_CMPINT (INT_MAX / 2, ==, mongoc_structured_log_opts_get_max_document_length (opts));
+
+   mongoc_structured_log_opts_destroy (opts);
+}
+
+int
+check_test_structured_log_env_defaults (void)
+{
+   // Skip testing env defaults if any options have been set externally
+   const char *expected_unset[] = {
+      "MONGODB_LOG_MAX_DOCUMENT_LENGTH",
+      "MONGODB_LOG_COMMAND",
+      "MONGODB_LOG_TOPOLOGY",
+      "MONGODB_LOG_SERVER_SELECTION",
+      "MONGODB_LOG_CONNECTION",
+      "MONGODB_LOG_ALL",
+   };
+
+   for (size_t i = 0u; i < sizeof expected_unset / sizeof expected_unset[0]; i++) {
+      const char *var = expected_unset[i];
+      char *value = test_framework_getenv (var);
+      bson_free (value);
+      if (value) {
+         MONGOC_DEBUG ("Skipping test because environment var '%s' is set", var);
+         return 0;
+      }
+   }
+   return 1;
+}
+
+void
+test_structured_log_env_defaults (void)
+{
```

Comment:
```suggestion
test_structured_log_env_defaults (void *ctx)
{
   BSON_UNUSED (ctx);

```
`AddFull*` test functions must accept `void *ctx` for function type consistency:
```
runtime error: call to function test_structured_log_env_defaults through pointer to incorrect function type 'void (*)(void)'
```